### PR TITLE
Feat: Add browserscript to paint header in duplicate tag color and automatically open duplicate profile in new tab

### PR DESCRIPTION
**Pull request by tmerten**
This script checks the dom on page load and on "candidate change" for a duplicate tag. If a tag is found it paints the entire header in duplicate tag color (as the tag can easily be overlooked)  and opens the candidate profile in a new tab, so that the profile loads in the background and duplicate checking requires one less click and can be performed more quickly.

Also add a simple test which checks if the header is correctly painted in duplicate tag color if a tag is found in the dom. 

_Unfortunately, I did not find a simple way to test the dynamic behavior and mock `GH_openInTab` from tampermonkey. However, the script is not destructive and does not change any data._

image

### CODE REVIEW

**Comment by tmerten** [comment]
While creating this PR I noticed that the eslint checks does not seem to be configured correctly in the project and thus formatting does not work. I did not fix this in this PR.

File: duplicate-check.user.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// ==UserScript==
+// @name         Open Candidate Profile in new Tab if Duplicate Badge is detected
+// @namespace    https://canonical.com/
+// @version      2024-11-25
```
Comment (nathanclairmonte) [inline duplicate-check.user.js:4]: Small change, just to keep this versioning in line with the other scripts. Otherwise it all looks good!

```suggestion
// @version      0.1.0
```

File: duplicate-check.user.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            );
+
+            if (duplicateTag) {
+                // we could get the link from the mutation but this also works for first load
+                let updatedLink = header.querySelector(
+                    'a[target="_blank"][href*="people"]'
+                );
+                // match greenhouse tag color
+                header.style.backgroundColor = "#f7c8b0";
+                GM_openInTab(updatedLink.href);
```
Comment (u-coskun) [inline duplicate-check.user.js:47]: I added the functionality to also open the duplicate profile in a new tab. Needs an additional grant though.
@nathanclairmonte @tmerten could you see if it works?
```
/* grab merge page url.
format: https://canonical.greenhouse.io/people/{candidateId}/merge?application_id={applicationId}
the new candidate page redesign alters how the url works, from "?application_id=" to "applications/"
*/
let mergeUrl = (updatedLink.href.includes("?application_id="))?
                updatedLink.href.replace("?application_id=", "/merge?application_id=") :
                updatedLink.href.replace("applications/", "/merge?application_id=");

// fetch merge page and open the duplicate profile in a new tab
GM_xmlhttpRequest({
    method: "GET",
    url: mergeUrl,
    onload: function(response) {
        let duplicateUrl = response.responseXML.getElementById("primary_merge_candidate").querySelector(
            'a[target="_blank"][href*="people"]'
        );
        GM_openInTab(duplicateUrl.href);
    }
});
```

File: duplicate-check.user.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+// @description  This script will color the entire header and open a new tab automatically if the candiate duplicate tag is detected. Saves time and reminds you to get merging.
+// @homepage     https://github.com/canonical/greenhouse-browser-scripts
+// @homepageURL  https://github.com/canonical/greenhouse-browser-scripts
+// @icon         https://icons.duckduckgo.com/ip3/greenhouse.io.ico
+// @homepage     https://github.com/canonical/greenhouse-browser-scripts
+// @match        https://canonical.greenhouse.io/applications/review/app_review*
+// @updateURL    https://raw.githubusercontent.com/canonical/greenhouse-browser-scripts/main/duplicate-check.user.js
+// @downloadURL  https://raw.githubusercontent.com/canonical/greenhouse-browser-scripts/main/duplicate-check.user.js
+// @supportURL   https://github.com/canonical/greenhouse-browser-scripts/issues
+// @grant        GM_openInTab
```
Comment (u-coskun) [inline duplicate-check.user.js:15]: These are the additional meta data
```
// @connect      https://canonical.greenhouse.io
// @grant        GM_xmlhttpRequest
```

File: duplicate-check.user.js (hunk 2 above)
Comment (tmerten) [reply duplicate-check.user.js:47]: I incorporated the changes and I think that I know what should happen (make sure to preload the profile's content, open merge screen directly instead of profile only) but it did not really work for me.
Maybe it's easiest if we can do a quick screen sharing session to talk it through.

File: duplicate-check.user.js (hunk 3 above)
Comment (tmerten) [reply duplicate-check.user.js:15]: As discussed, let's merge this as is and then you can push these changes as a follow up MP so that it's credited to you.
We should update the description though that it will open both profiles automatically :)

If I'm not mistaken this is the first time I get a contribution before my contribution even landed. Thanks for that :muscle:

**Review by nathanclairmonte — APPROVED** [review]
